Content.Server/Polymorph/Systems/PolymorphSystem.cs

```diff
@@ -406,3 +413,16 @@ public void RemovePolymorphAction(ProtoId<PolymorphPrototype> id, Entity<Polymor
             _actions.RemoveAction(target, val);
     }
 }
+
+/// <summary>
+/// Raised directed on an entity before polymorphing it.
+/// Cancel to stop the entity from being polymorphed.
+/// </summary>
+/// <param name="Target">The entity that is being polymorphed into.</param>
+[ByRefEvent]
+public record struct BeforePolymorphedEvent(Entity<PolymorphedEntityComponent?> Target)
+{
+    public readonly Entity<PolymorphedEntityComponent?> Target = Target;
+
+    public bool Canceled;
+}
```

Comment:
move to shared _DV and change target to just EntityUid since you dont set the component part anywhere

```cs
[ByRefEvent]
public record struct BeforePolymorphedEvent(EntityUid Target, bool Cancelled = false);
```

(you also misspelled cancelled)